src/GafferImage/Sampler.cpp

```diff
@@ -73,6 +73,10 @@ Sampler::Sampler( const GafferImage::ImagePlug *plug, const std::string &channel
 		// all bounds checking.
 		m_boundingMode = -1;
 	}
+	else if( BufferAlgo::empty( m_dataWindow ) )
```

Comment:
Yep, that's all exactly correct.

The motivation for this was mostly at the beginning when I was trying to avoid writing visitPixels, and trying to wring out every drop of performance without the change in API - now that I've decided visitPixels is necessary, it's less crucial, but I think it did help save about a percent on some test, so it probably doesn't hurt to keep it.